docs/sphinx/examples/cpp/quantinuum

```diff
@@ -0,0 +1,35 @@
+// Compile and run with:
+// ```
+// nvq++ --target quantinuum quantinuum.cpp -o out.x && ./out.x
+// ```
+// Assumes a valid set of credentials have been stored. 
+
+#include <cudaq.h>
+
+// Define a simple quantum kernel to execute on Quantinuum.
+struct ghz {
+  // Maximally entangled state between 5 qubits.
+  auto operator()() __qpu__ {
+    cudaq::qreg q(5);
+    h(q[0]);
+    for (int i = 0; i < 5 - 1; i++) {
+      x<cudaq::ctrl>(q[i], q[i + 1]);
+    }
+    mz(q);
+  }
+};
+
+int main() {
+  // Submit to Quantinuum synchronously. E.g, wait for the job
+  // result to be returned before proceeding. 
+  auto counts = cudaq::sample(ghz{});
+  counts.dump();
+
+  // Submit to Quantinuum asynchronously. E.g, continue executing
+  // code in the file until the job has been returned.
+  auto future = cudaq::sample_async(ghz{});
+  // ... classical code to execute in the meantime ...
+  auto async_counts = future.get();
+  // TODO: We can also persist the future results to a file.
```

Comment:
Let's get rid of the todo before merging. I suggest to just make it a comment, or show how to write it to a file.

Reply:
I agree, let's add the code now.